From db32a4d0687974c03dbd25d951d6a795d5b832ef Mon Sep 17 00:00:00 2001
From: claxten10 <claxten10@gmail.com>
Date: Tue, 11 Nov 2025 18:35:00 +0100
Subject: [PATCH 21/35] res: Replace fixed dp with a percentage of screen

* Inspired from T transitions, where instead of setting a fixed 96dp, it used
  a percentage of screen.

test: - open settings app
 - go to a menu
 - transition should be quick and smooth not long and choppy
Change-Id: If9675f93ec795849477a90298f32c2bcd4778a3e
Signed-off-by: claxten10 <claxten10@gmail.com>
---
 core/res/res/anim/activity_close_enter.xml | 8 ++++----
 core/res/res/anim/activity_close_exit.xml  | 6 +++---
 core/res/res/anim/activity_open_enter.xml  | 6 +++---
 core/res/res/anim/activity_open_exit.xml   | 8 ++++----
 4 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/core/res/res/anim/activity_close_enter.xml b/core/res/res/anim/activity_close_enter.xml
index 22a1dd691..8cb43c743 100644
--- a/core/res/res/anim/activity_close_enter.xml
+++ b/core/res/res/anim/activity_close_enter.xml
@@ -31,7 +31,7 @@
         android:duration="450" />
 
     <translate
-        android:fromXDelta="-96dp"
+        android:fromXDelta="-10%"
         android:toXDelta="0"
         android:fillEnabled="true"
         android:fillBefore="true"
@@ -43,13 +43,13 @@
     <extend
         android:fromExtendLeft="0"
         android:fromExtendTop="0"
-        android:fromExtendRight="96dp"
+        android:fromExtendRight="10%"
         android:fromExtendBottom="0"
         android:toExtendLeft="0"
         android:toExtendTop="0"
-        android:toExtendRight="96dp"
+        android:toExtendRight="10%"
         android:toExtendBottom="0"
         android:interpolator="@interpolator/fast_out_extra_slow_in"
         android:startOffset="0"
         android:duration="450" />
-</set>
\ No newline at end of file
+</set>
diff --git a/core/res/res/anim/activity_close_exit.xml b/core/res/res/anim/activity_close_exit.xml
index a6710490d..f807c26dd 100644
--- a/core/res/res/anim/activity_close_exit.xml
+++ b/core/res/res/anim/activity_close_exit.xml
@@ -32,7 +32,7 @@
 
     <translate
         android:fromXDelta="0"
-        android:toXDelta="96dp"
+        android:toXDelta="10%"
         android:fillEnabled="true"
         android:fillBefore="true"
         android:fillAfter="true"
@@ -41,11 +41,11 @@
         android:duration="450" />
 
     <extend
-        android:fromExtendLeft="96dp"
+        android:fromExtendLeft="10%"
         android:fromExtendTop="0"
         android:fromExtendRight="0"
         android:fromExtendBottom="0"
-        android:toExtendLeft="96dp"
+        android:toExtendLeft="10%"
         android:toExtendTop="0"
         android:toExtendRight="0"
         android:toExtendBottom="0"
diff --git a/core/res/res/anim/activity_open_enter.xml b/core/res/res/anim/activity_open_enter.xml
index f3172e4ca..1674dab30 100644
--- a/core/res/res/anim/activity_open_enter.xml
+++ b/core/res/res/anim/activity_open_enter.xml
@@ -30,7 +30,7 @@
         android:duration="83" />
 
     <translate
-        android:fromXDelta="96dp"
+        android:fromXDelta="10%"
         android:toXDelta="0"
         android:fillEnabled="true"
         android:fillBefore="true"
@@ -39,11 +39,11 @@
         android:duration="450" />
 
     <extend
-        android:fromExtendLeft="96dp"
+        android:fromExtendLeft="10%"
         android:fromExtendTop="0"
         android:fromExtendRight="0"
         android:fromExtendBottom="0"
-        android:toExtendLeft="96dp"
+        android:toExtendLeft="10%"
         android:toExtendTop="0"
         android:toExtendRight="0"
         android:toExtendBottom="0"
diff --git a/core/res/res/anim/activity_open_exit.xml b/core/res/res/anim/activity_open_exit.xml
index d84827b83..d98ce93f6 100644
--- a/core/res/res/anim/activity_open_exit.xml
+++ b/core/res/res/anim/activity_open_exit.xml
@@ -31,7 +31,7 @@
 
     <translate
         android:fromXDelta="0"
-        android:toXDelta="-96dp"
+        android:toXDelta="-10%"
         android:fillEnabled="true"
         android:fillBefore="true"
         android:fillAfter="true"
@@ -42,13 +42,13 @@
     <extend
         android:fromExtendLeft="0"
         android:fromExtendTop="0"
-        android:fromExtendRight="96dp"
+        android:fromExtendRight="10%"
         android:fromExtendBottom="0"
         android:toExtendLeft="0"
         android:toExtendTop="0"
-        android:toExtendRight="96dp"
+        android:toExtendRight="10%"
         android:toExtendBottom="0"
         android:interpolator="@interpolator/fast_out_extra_slow_in"
         android:startOffset="0"
         android:duration="450" />
-</set>
\ No newline at end of file
+</set>
-- 
2.52.0

